### ISSUE TITLE
Fix array indexing and slicing

### ISSUE THREAD

**Issue by namannimmo10** [lcompilers/lpython#332]
```python
def main0():
    arr: i32[4]
    arr[0] = 1
    arr[1] = 0
    arr[2] = -4
    print(arr[:2])
    print(arr[1])
    print(arr[2])

main0()
```
This code gives the following output:
```console
-4
0
-4
```
We should fix this bug.

**Comment by czgdp1807** [comment]
This might be due to the fact that array indexing works very differently in Python and Fortran. The array slice pass I wrote works correctly for LFortran. For LPython we need to do some debugging in https://github.com/lcompilers/lpython/blob/main/src/libasr/pass/arr_slice.cpp. Should be a medium level issue.

**Comment by certik** [comment]
Yes, we must make sure the frontend creates the ASR correctly for slices.
